frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/traverse/TraverseAction.java

```diff
@@ -0,0 +1,126 @@
+
+  @Override
+  @UIEffect
+  protected void onUpdate(AnActionEvent anActionEvent) {
+    super.onUpdate(anActionEvent);
+    val presentation = anActionEvent.getPresentation();
+    if (!presentation.isEnabledAndVisible()) {
+      return;
+    }
+    presentation.setDescription(getNonHtmlString("action.GitMachete.TraverseAction.description.text"));
+
+    val graphTable = getGraphTable(anActionEvent);
+    val repositorySnapshot = graphTable.getGitMacheteRepositorySnapshot();
+    val branchLayout = repositorySnapshot != null ? repositorySnapshot.getBranchLayout() : null;
+
+    if (branchLayout == null || branchLayout.getRootEntries().isEmpty()) {
+      presentation.setEnabled(false);
+      presentation.setDescription(getNonHtmlString("action.GitMachete.TraverseAction.description.empty-layout"));
+      return;
+    }
+  }
+
+  @Override
+  @UIEffect
+  public void actionPerformed(AnActionEvent anActionEvent) {
+    val gitRepository = getSelectedGitRepository(anActionEvent);
+    val graphTable = getGraphTable(anActionEvent);
+    val repositorySnapshot = graphTable.getGitMacheteRepositorySnapshot();
+    val branchLayout = repositorySnapshot != null ? repositorySnapshot.getBranchLayout() : null;
+    val project = getProject(anActionEvent);
+
+    boolean yesNoResult = true;
+    if (branchLayout != null && branchLayout.getRootEntries().nonEmpty() && gitRepository != null) {
+      if (PropertiesComponent.getInstance(project).getBoolean(SHOW_TRAVERSE_INFO, /* defaultValue */ true)) {
+
+        val traverseInfoDialog = new InfoDialog(
+            project,
+            getString("action.GitMachete.TraverseAction.dialog.traverse-approval.title"),
+            getString("action.GitMachete.TraverseAction.dialog.traverse-approval.text.HTML"),
+            SHOW_TRAVERSE_INFO);
+
+        yesNoResult = traverseInfoDialog.showAndGet();
+      }
+
+      if (yesNoResult) {
+        var firstEntry = branchLayout.getRootEntries().headOption().getOrNull();
+        if (firstEntry != null) {
+          val firstEntryName = firstEntry.getName();
+          checkoutAndTraverse(gitRepository, graphTable, firstEntryName);
+        }
+      }
+    }
+  }
+
+  private void checkoutAndTraverse(GitRepository gitRepository, BaseEnhancedGraphTable graphTable, String branchName) {
+    log().debug(() -> "Queuing '${branchName}' branch checkout background task");
+
+    @UI Runnable traverseRunnable = () -> traverse(gitRepository, graphTable, branchName);
+    Runnable repositoryRefreshRunnable = () -> graphTable.queueRepositoryUpdateAndModelRefresh(traverseRunnable);
+    Runnable callInAwtLater = () -> ModalityUiUtil.invokeLaterIfNeeded(NON_MODAL, repositoryRefreshRunnable);
+    val gitBrancher = GitBrancher.getInstance(gitRepository.getProject());
+    val repositories = Collections.singletonList(gitRepository);
+    gitBrancher.checkout(/* reference */ branchName, /* detach */ false, repositories, callInAwtLater);
```

Comment:
Similarly to how `git machete traverse` operates, I'd skip checking out the branches for which there's nothing to be done anyway

Reply:
done! however there was one advantage of checking out the branch - the branch under operation was always indicated in the graph table this way. but I agree - it's better to avoid checkout if not needed

Reply:
As discussed, let's also check out the branch to be pushed/slid out (not only the ones to be rebased)

Reply:
Huh in the option 2... you mean
> do checkout, slide out, but do not allow for delete

right? I'd go for that, that's what CLI does anyway in `git machete traverse` (it still doesn't have branch deletion on slide-out implemented :trollface:)

Reply:
ok, that's the easiest and actually done by always checking out the branch during the traverse